src/components/pipelines-overview/utils.ts

```diff
@@ -239,3 +263,35 @@ export const getFilter = (date, parentName, kind): string => {
   }
   return filter.join(' && ');
 };
+
+export const useQueryParams = (param) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const {
+    key,
+    value,
+    setValue,
+    defaultValue,
+    options,
+    displayFormat,
+    loadFormat,
+  } = param;
+  React.useEffect(() => {
+    if (searchParams.has(key)) {
+      const paramValue = searchParams.get(key);
+      if (!options || options[paramValue])
+        setValue(loadFormat ? loadFormat(paramValue) : paramValue);
+    }
+  }, []);
+
+  React.useEffect(() => {
+    setSearchParams((prevParams) => {
+      const newParams = new URLSearchParams(prevParams);
+      if (value && value != defaultValue)
```

Comment:
even for the default values it should add params in the URL.

Reply:
ye had some queries on that, but ya should be good. Need to deal with concurrent initialization of the same by all of the fields.